### ISSUE TITLE
Correct way to handle text abutting citation?

### ISSUE THREAD

**Issue by jcushman** [freelawproject/eyecite#52]
What do we think is the right thing to do with text like this?

```
In [2]: get_citations("foo1 U.S. 1bar")
Out[2]: [FullCaseCitation(token=CitationToken(data='1 U.S. 1bar', start=3, end=14, volume='1', reporter='U.S.', page='1bar'...
```

We're currently permissive about finding cites with abutting characters, which I can imagine being overinclusive with reporter strings like "foo1 Or 2bar" or "foo1 P 2bar" or "foo1 mt 2bar" etc.

I guess this breaks down into a few separate things:

1. Should there be a defined list of characters that can come before the start of a cite, like `(^| )`? I'm thinking probably yes?
1. Should there be a defined list of characters that can come after a cite? This seems harder to make complete, but might be like `($|[ ,.;])` and a few more along those lines. Less sure about that one.
1. Should we narrow the definition of page numbers so that example would come back as `page='1'`? We currently capture `1bar` because of the page number regex `\d{1,6}[-]?[a-zA-Z]{1,6}`, which is there to handle `A special Connecticut or Illinois number. E.g., "13301-M"`. So we could add that regex specifically to CT and IL in reporters-db and avoid capturing random cruft in other reporters. I'm leaning toward that being a good idea.

**Comment by mlissner** [comment]
I guess our bias should be towards underinclusive correctness. If we start from there, I think the best options are numbers 1 and 3. Option 1, tightening what can come before a citation, seems easy. Option 2, tightening what comes after, seems hard. Option 3, tightening our current regex and then dialing in special cases, seems not too bad either.

**Comment by mlissner** [comment]
This is fixed via #61, right @jcushman ?

**Comment by jcushman** [comment]
Yep!
